include/nbl/asset/IAsset.h

```diff
@@ -242,6 +248,26 @@ class IAsset : virtual public core::IReferenceCounted
 			//_NBL_DEBUG_BREAK_IF(imm);
 			return imm;
 		}
+		//! compares two objects for matching types and, in derived types, compares member variable values. Common code for IAsset::equals, IAsset::canBeRestoredFrom
+		virtual bool compatible(const IAsset* _other) const {
+			_other->getAssetType() == getAssetType();
+		}
```

Comment:
follow this pattern
```cpp
inline bool compatible() const
{
    if (getAssetType()==_other->getAssetType())
        return false;
    return compatible_impl();
}
```

where `_impl` is a pure virtual method.

Reply:
same thing for `equals`

```cpp
inline bool equals(const IAsset* _other) const
{
   if (compatible(_other))
        return equals_impl(_other);
   return false;
}
```

Reply:
https://discord.com/channels/593902898015109131/1078226704092381194/1083069244650045600